builder/qemu/config.go

```diff
@@ -288,6 +353,11 @@ type Config struct {
 	// The type of machine emulation to use. Run your qemu binary with the
 	// flags `-machine help` to list available types for your system. This
 	// defaults to `pc`.
+	//
+	// NOTE: when booting a UEFI machine with Secure Boot enabled, this has
+	// to be a q35 derivative.
```

Comment:
Should there be some validation or warning in Prepare to let the user know that they have enabled EFI with a non-q35 derivative machine type?

 I'm not sure if this is a simple check. So I would be cook if you tell me there are too many combinations to validate; its better for the user to see the failure 😉

Reply:
EFI on a non-q35 machine is fine, the problem is Secure-Boot, which is only supported on q35 machines (can't fine the docs to refer to that, but other machine types AFAIK will result in a blank screen).
Not sure we can reliably validate that, maybe by checking the name for the OVMF_*fd, but that's a stretch as a user might rename them, and the check will not bump the error.